src/couch_proc_manager.erl

```diff
@@ -436,9 +436,8 @@ assign_proc(#client{}=Client, #proc_int{client=undefined}=Proc) ->
     assign_proc(Pid, Proc).
 
 
-return_proc(#state{} = State, #proc_int{} = ProcInt) ->
-    #proc_int{pid = Pid, lang = Lang} = ProcInt,
-    NewState = case is_process_alive(Pid) of true ->
+return_proc(#state{} = State, #proc_int{pid = Pid} = ProcInt) ->
+    case is_process_alive(Pid) of true ->
         case ProcInt#proc_int.t0 < State#state.threshold_ts of
```

Comment:
But this is pretty much the same as it is right now, we are always [flushing](https://github.com/apache/couchdb-couch/blob/master/src/couch_proc_manager.erl#L455) in return_proc, but in `false` clause it's meaningless, because state's counts hasn't been updated.

Out of curiosity, how this generational collection controlled? I understand what it works, but is there anything that uses it or is it just for manual `couch_proc_manager:reload()`?